### PR TITLE
Fixed idempotency issue by first checking if table does not exist

### LINKED ISSUE #1033: [BUG]: Create Table command in AppStack.ts not idempotent.
### Operating system

Windows 11

### Bug description

Create Table command in AppStack.ts throws an exception if the specified table already exists in DynamoDB. Prevents SST from starting. In other words, the command is not idempotent. We should check if the table exists first, and if not, create it then.

### Steps to reproduce

Un-comment the 
```
const trainspaceRunTable = new Table(...); 
```
lines of code, and then type dlp-cli start.

It should complain about the specified table already existing, and then fail to start.

### PR DESCRIPTION
**Pull request by alantao912**
# Fixed idempotency issue by first checking if table does not exist

Github Issue Number Here: <#1033>
**What user problem are we solving?**
AppStack.ts does not start because trainspace table already exists.

**What solution does this PR provide?**
Checks if the table does not already exist before creating it again.

**Testing Methodology**
Typed dlp-cli serverless start and it worked.

**Any other considerations**

Issue may have been fixed by something else.

### CODE REVIEW

File: serverless/stacks/AppStack.ts [hunk 1]
```diff
@@ -11,26 +12,29 @@ export function AppStack({ stack }: StackContext) {
...
+    trainspaceRunTable = new Table(stack, "trainspace-run", {
+      fields: {
+        runId: "string",
+        trainspaceId: "string",
+        userId: "string",
+        timestamp: "string",
+        resultCsvUri: "string",
+        modelPtUri: "string",
+        onnxUri: "string",
+        confusionMatrixUri: "string",
```
Comment (karkir0003) [inline serverless/stacks/AppStack.ts]: will this code work for if I want to update fields of a dynamo table?

**Review by karkir0003 — COMMENTED** [review]
added a question

**Comment by farisdurrani** [comment]
Minor: can you please run Prettier for all changed files? There are some aesthetic issues; for example, each file should end with a blank line. 

We'll need to update our checks to also check /serverless in addition to /frontend

File: serverless/packages/functions/src/dbutils/table_exists.ts [hunk 2]
```diff
@@ -0,0 +1,6 @@
+import { get_all_tablenames } from "./get_all_tablenames";
+
+export async function table_exists(table_name: string) : Promise<boolean> {
+    const tableNames: string[] = await get_all_tablenames();
+    return tableNames.indexOf(table_name) !== -1;
+}
```
Comment (farisdurrani) [inline serverless/packages/functions/src/dbutils/table_exists.ts:24]: Minor: can you please run Prettier for all changed files? There are some aesthetic issues; for example, each file should end with a blank line.

We'll need to update our checks to also check /serverless in addition to /frontend

**Comment by karkir0003** [comment]
@alantao912 can you ask in sst discord best practice for checking if a table exists before creating a new one. ask if sst handles that check sort of under the hood

**Comment by alantao912** [comment]
> @alantao912 can you ask in sst discord best practice for checking if a table exists before creating a new one. ask if sst handles that check sort of under the hood

image
I already tried </3 :(

**Comment by karkir0003** [comment]
> > @alantao912 can you ask in sst discord best practice for checking if a table exists before creating a new one. ask if sst handles that check sort of under the hood
> 
> image I already tried </3 :(

is there a way we can tag someone who maintains SST in this thread @alantao912

File: serverless/stacks/AppStack.ts [hunk 3]
```diff
@@ -22,15 +27,34 @@ export function AppStack({ stack }: StackContext) {
...
+            confusionMatrixUri: "string",
+            aucRocUri: "string"
+          },
+          primaryIndex: {
+            partitionKey: "runId",
+          },
+          globalIndexes: {
+            "TrainspaceIndex": { partitionKey: "trainspaceId", sortKey: "timestamp"},
+            "UserIndex": {partitionKey: "userId"}
+          },
```
Comment (karkir0003) [inline serverless/stacks/AppStack.ts:52]: What if I want to update a GSI or User index. Can we update the PR to be resilient for those cases? @alantao912

**Review by karkir0003 — COMMENTED** [review]
added a quick comment @alantao912

**Review by farisdurrani — APPROVED** [review]

File: serverless/stacks/AppStack.ts (hunk 3 above)
Comment (alantao912) [reply serverless/stacks/AppStack.ts:52]: I don't think the current approach is correct. I'm looking into other solutions.

**Comment by karkir0003** [comment]
@alantao912 status on this?

**Pull request closed without merging**